### PR TITLE
[CS2113-T14-3] UniTasker

### PR DESCRIPTION
**Pull request by marken9**
UniTasker is a one-stop CLI planner for university students to quickly manage and view everything studies-related — tasks and courses.

### CODE REVIEW

File: docs/pictures/ReorderTodoSequenceDiagram.png
Comment (yueyang1111) [inline docs/pictures/ReorderTodoSequenceDiagram.png:1]: Just some minor things? But some of the method names do not match the actual implementation (e.g., execute(line), getCategories()), which may cause confusion for the readers

File: docs/DeveloperGuide.md
Comment (yueyang1111) [inline docs/DeveloperGuide.md:1]: The Storage class diagram is  accurate and reflects the main methods and responsibilities well. It was also well supported by the description in the dg.

Comment (yueyang1111) [inline docs/DeveloperGuide.md:1]: The delete marked and event mapping has good explanation and design consideration, which made it easy to follow!

Comment (yueyang1111) [inline docs/DeveloperGuide.md:1]: Not sure if its intentional, but currently Key Design Consideration the code are rendered as `list event /all` with the ` `.

**Review by yueyang1111 — COMMENTED** [review]
Overall the developer guide is very neat and well structured!

File: docs/DeveloperGuide.md
Comment (yueyang1111) [inline docs/DeveloperGuide.md:1]: Can add alternative considerations and more test cases maybe?

Comment (yueyang1111) [reply docs/DeveloperGuide.md:1]: its rendered with `

Comment (yueyang1111) [inline docs/DeveloperGuide.md:1]: Overall the developer guide is very easy to follow with good diagrams like the different class and sequence diagram to support the description!

Comment (HX2003) [inline docs/DeveloperGuide.md:1]: Overall a well rounded developer guide with sufficient diagrams. As a suggestion for improvement, I noticed some of the images were blurry which makes it harder to read, which is compounded by the fact that the text is very small. Perhaps vector format would be more suitable.

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -2,37 +2,498 @@
...
+
+The bulk of the app's work is done by the following components:
+- Command: Responsible for executing user instructions correctly
+- UI: Responsible for displaying messages to the user.
+- AppContainer: Acts as a central container for sharing application data.
+- Storage: Responsible for saving and loading data from local files.
+
+**How the architecture components interact with each other:**
+
+![ReorderCommand](pictures/ReorderTodoSequenceDiagram.png)
```
Comment (HX2003) [inline docs/DeveloperGuide.md:41]: In your explanation, it was mentioned that UniTasker passes it to `CommandParser`, yet in the diagram I can't find a reference to it.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -2,37 +2,498 @@
...
+
+3. ListCommand calls `CategoryList.getAllEvents()` with flags determining the view (e.g., collapsed vs. expanded).
+
+4. Mapping Generation: As the `CategoryList` iterates through categories and events to build the display string, it simultaneously populates the `ActiveDisplayMap`.
+
+5. Each displayed line is assigned a UI Index, which is mapped to an `EventReference` containing the (categoryIndex, eventIndex).
+
+6. The formatted string is sent to `GeneralUi` for display, and the `currentView` is updated to allow subsequent delete or edit commands to reference the correct map.
+
+**Sequence Diagram for `list event` command**
```
Comment (HX2003) [inline docs/DeveloperGuide.md:265]: In the diagram, `new ListCommand(...)` is called, but there is no corresponding activation bar to suggest the creation of it.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -2,37 +2,498 @@
...
+- If yes throw an OverlapEventException, otherwise all validators have been passed and task is added successfully
+
+
+### Feature: Course Tracker
+
+The Course Tracker feature allows students to manage their courses and track their assessment scores and weightages.
+
+**Course Class Diagram**
+
+![Course Class Diagram](pictures/CourseClassDiagram.png)
```
Comment (HX2003) [inline docs/DeveloperGuide.md:416]: I believe the lecturer said that the default plantuml styling for class diagram is non-compliant for the class name and the visibility, I think there's a way to switch it.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -2,37 +2,498 @@
...
+|       0       |    2    |                   (0,3)                    | CS2113 lecture  |
+|       1       |    1    |                   (1,0)                    |   yoga lesson   |
+
+
+
+**2. Polymorphism and Time-Based Tasks**
+
+UniTasker treats deadlines and events differently from todos to enable advanced scheduling features:
+- **Inheritance**: A shared interface allows the `Calendar` class to store and query deadlines and events polymorphically. This enables range queries (e.g., "show all tasks this week") without the `Calendar` needing to know the specific task type.
+- **Efficient Lookups**: The `Calendar` uses a `TreeMap<LocalDate, List<Task>>`. This structure provides $O(\log n)$ lookup times for specific dates and efficient range-based sub-maps, avoiding the need to iterate through the entire global task list.
```
Comment (HX2003) [inline docs/DeveloperGuide.md]: Nice rationale and explanation for how your design was implemented

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (HX2003) [inline docs/DeveloperGuide.md:265]: The activation bar for list command seems to extend after stuff was returned.

**Review by HX2003 — COMMENTED** [review]
Overall a well rounded developer guide with sufficient diagrams. As a suggestion for improvement, I noticed some of the images were blurry which makes it harder to read, which is compounded by the fact that the text is very small. Perhaps vector format would be more suitable.

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (HX2003) [reply docs/DeveloperGuide.md:416]:
```
skinparam classAttributeIconSize 0
hide circle
```

should do the job

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -2,37 +2,498 @@
...
+- Storage: Responsible for saving and loading data from local files.
+
+**How the architecture components interact with each other:**
+
+![ReorderCommand](pictures/ReorderTodoSequenceDiagram.png)
+
+1. User enters command in terminal
+2. UniTasker reads the input and passes it to CommandParser
+3. CommandParser identifies the command type and creates a corresponding Command object
+4. The Command object calls its execute() method with the AppContainer instance
```
Comment (WangZX2001) [inline docs/DeveloperGuide.md:48]: I think there is a missing attribute here in the documentation for the execute() method. In the UML diagram, it is shown as execute(line)

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (WangZX2001) [reply docs/DeveloperGuide.md:41]: Agree with this point here, the diagram is missing the CommandParser

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -2,37 +2,498 @@
...
+- task management logic remains encapsulated within the model
+- bulk operations can be applied consistently across all task types
+
+This design improves modularity and keeps command logic simple, while allowing `CategoryList` to coordinate operations affecting all three task lists.
+
+### Deadline Class Diagram
+
+The figure below illustrates the relationship between Deadline class and the following classes: Task, Timed, Calendar, DateUtils, DeadlineList, TaskList. 
+
+![Deadline Class Diagram](pictures/deadlineClassDiagram.png)
```
Comment (WangZX2001) [inline docs/DeveloperGuide.md:203]: I think there are too many relationship being shown in the deadline class diagram and only the methods and attributes of the deadline class is being explained. Maybe you can consider focusing on the deadline class instead so that the diagram will look more organised and easier to interpret.

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -2,37 +2,498 @@
...
+2. Input is parsed and `AddEventCommand` is created.
+3. `AddEventCommand` calls `CategoryList` functions to add events.
+4. If non-recurring:
+   - `EventList` `add(Event)` is called
+5. If recurring:
+   - `EventList` `addRecurringWeeklyEvent(addRecurringWeeklyEvent(event, calendar, date, months)` is called to generate and group weekly events.
+6. Event(s) are stored in `EventList`, persisted by `Storage`, and reflected in the `Calendar`
+
+**Sequence Diagram for `add event` command**
+![AddEvent Sequence Diagram](pictures/AddRecurringNonRecurringEvent.png)
```
Comment (WangZX2001) [inline docs/DeveloperGuide.md]: Good to see that there is a lot of details in the Add Event command sequence diagram. However, I think the diagram is too complicated and the font of the words in the diagram is also quite difficult to see. May be you can consider using ... to represent some of the parameters of the method or use pseudo code instead of typing the whole method out?

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -2,37 +2,498 @@
...
+The following class diagram shows the structure of the undo feature:
+
+![UndoClassDiagram](pictures/UndoClassDiagram.png)
+
+**Key Design Considerations**
+
+- `UniTasker` maintains a `Stack<Command>` to track undoable commands
+- `Command` interface exposes default `undo()` and `isUndoable()` methods so existing commands are unaffected
+- Only course commands that modify data are pushed to the stack (`add`, `delete`, `add-assessment`)
+- Undo history is cleared on app exit
```
Comment (WangZX2001) [inline docs/DeveloperGuide.md:448]: Good to see that you are using both class diagram and sequence diagram to illustratethe Undo Features. But I think you can also consider adding a detailed workthrough or explanation of this undo command to help me better understand about its implementation details.

**Review by WangZX2001 — COMMENTED** [review]
Good to see that you guys have put in a lot of efffort in this developer guide. There are a good amount of class diagram, object diagram as well as sequence diagram used to illustrate the application implementation. There is also good amount of explanation on the detail workthrough of how each command works. Overall, I think this is a well developed guide. A small suggestion I have is to may be consider removing some of the less important attributes and details in the class and sequence diagram because I realise that some of these diagrams contain too much information and details that are a bit overwhelming.

File: docs/pictures/CategoryListClassDiagram.png
Comment (RishabhShenoy03) [inline docs/pictures/CategoryListClassDiagram.png:1]: Comments are very helpful in understanding the diagram

File: docs/pictures/DateUtilsSequence.png
Comment (RishabhShenoy03) [inline docs/pictures/DateUtilsSequence.png:1]: Method signatures don't exactly match their implementation, could consider changing them to be more reflective of the code?

File: docs/pictures/UndoSequenceDiagram.png
Comment (RishabhShenoy03) [inline docs/pictures/UndoSequenceDiagram.png:1]: Execute reverse operation is well explained in the comment

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -2,37 +2,498 @@
...
+
+![CommandClassDiagram](pictures/CommandClassDiagram.png)
+
+The `Command` component handles the execution of user commands. Each supported command is represented by a separate command class implementing the common `Command` interface.
+
+The component consists of:
+- `CommandParser`, which maps raw user input to a concrete command object
+- `Command`, which defines the common `execute(AppContainer container)` method
+- Concrete command classes such as `AddCommand`, `DeleteCommand`, and so on.
+- `CommandSupport`, which provides shared helper methods such as data saving and category index retrieval
```
Comment (RishabhShenoy03) [inline docs/DeveloperGuide.md:117]: Not really understanding the idea of category index, are they arbitrary and self-defined? Maybe some explanation would be helpful for users

File: docs/DeveloperGuide.md
Comment (RishabhShenoy03) [inline docs/DeveloperGuide.md:1]: Overall, nice DG with some minor typos here and there. Sufficient diagrams with nice comments and explanations, but method calls and signatures could use some updating. Would also benefit from larger text in the diagrams so readers wouldn't need to zoom in for each image. Great job!

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -2,37 +2,498 @@
...
+| categoryIndex | uiIndex | EventReference (categoryIndex, eventIndex) |   Description   |
+|:-------------:|:-------:|:------------------------------------------:|:---------------:|
+|       0       |    1    |                   (0,0)                    |  consultation   |
+|       0       |    2    |                   (0,1)                    | CS2113 tutorial |
+|       0       |    3    |                   (0,2)                    |     meeting     |
+|       0       |    4    |                   (0,3)                    | CS2113 lecture  |
+|       1       |    1    |                   (1,0)                    |   yoga lesson   |
+
+
+    `list recurring`
```
Comment (RishabhShenoy03) [inline docs/DeveloperGuide.md]: Tables make it clear how the output will be formatted, helps me understand how to format my inputs better as well for the description and categoryIndex

File: docs/DeveloperGuide.md [hunk 11]
```diff
@@ -2,37 +2,498 @@
...
+**Workflow for Add Event command**
+1. User enters either:
+   - `add event <categoryIndex> <description> /from <startDateTime> /to <endDateTime>`
+   - `add recurring <categoryIndex> weekly event <description> /from <day> <time> /to <day> <time> /(date or month) <dateOrMonth>`
+2. Input is parsed and `AddEventCommand` is created.
+3. `AddEventCommand` calls `CategoryList` functions to add events.
+4. If non-recurring:
+   - `EventList` `add(Event)` is called
+5. If recurring:
+   - `EventList` `addRecurringWeeklyEvent(addRecurringWeeklyEvent(event, calendar, date, months)` is called to generate and group weekly events.
```
Comment (alsonsim) [inline docs/DeveloperGuide.md]: In the Add Event Command workflow, the behaviour states: "EventList addRecurringWeeklyEvent(addRecurringWeeklyEvent(event, calendar, date, months)) is called". This appears to be a copy-paste error where the method name is nested inside itself. Should this read addRecurringWeeklyEvent(event, calendar, date, months) once? A duplicated method name in the DG could mislead a future developer tracing the call chain.

File: docs/DeveloperGuide.md [hunk 12]
```diff
@@ -2,37 +2,498 @@
...
+
+1. User types `delete event <categoryIndex> <uiIndex>` which is parsed by `CommandParser` class to
+   create a `DeleteCommand` object
+2. In `DeleteCommand` under 'event' uiIndex is parsed and used as an index in the list of `EventReference` objects to
+   get the particular `EventReference` object
+3. Event is then deleted using `EventReference.categoryIndex` and `EventReference.eventIndex` if it is non-recurring. If it is recurring it will prompt user to use `list occurrence`
+4. Changes are updated in `Storage` class and `Calendar` object
+
+Note:
+- `list event` must be called before `delete event <categoryIndex> <uiIndex>` to populate the map correctly
```
Comment (alsonsim) [inline docs/DeveloperGuide.md:282]: The Delete Event Command section mentions: "Note: list event must be called before delete event <categoryIndex> <uiIndex> to populate the map correctly." This is a significant usability constraint — the delete command silently depends on a prior list command. Should this prerequisite be elevated from a footnote to a clearly labelled Precondition at the top of the section, rather than buried in the Notes? A future developer or tester reading only the delete section could miss this entirely.

File: docs/DeveloperGuide.md (hunk 8 above)
Comment (alsonsim) [inline docs/DeveloperGuide.md:448]: The Feature: Undo section states "Only course commands that modify data are pushed to the stack (add, delete, add-assessment)" and "Undo history is cleared on app exit." However, the target user profile emphasises managing todos, deadlines, and events as core workflows. Should the DG explain the design rationale for limiting undo to course commands only, and explicitly excluding task commands? Without this explanation, a future developer may incorrectly assume undo is meant to be extended to all commands, or may not understand why the scope is restricted.

File: docs/DeveloperGuide.md [hunk 13]
```diff
@@ -2,37 +2,498 @@
...
+UniTasker is designed for university students who need to manage multiple courses, assignments, deadlines, and personal tasks simultaneously. These users often:
+- juggle academic responsibilities across different modules, each with varying deadlines, 
+priorities, and schedules. 
+- They require a simple and efficient system to organize their tasks,
+keep track of coursework, and stay on top of deadlines.
+- prefer a fast, keyboard-driven interface over GUI-heavy applications
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
```
Comment (alsonsim) [inline docs/DeveloperGuide.md:19]: In the Value proposition section, the second paragraph reads: "This helps students to stay on top of their workload and focus on completing their academic responsibilities." The paragraph before it ends mid-thought: "These users often: juggle academic responsibilities... They require a simple..." — this switches between a bullet-implied list and a full sentence without a clean structure. Should this be reformatted into either a proper bulleted list under "These users often:" or two separate sentences? The current phrasing is grammatically inconsistent and harder to read.

File: docs/DeveloperGuide.md [hunk 14]
```diff
@@ -2,37 +2,498 @@
...
+| v2.0    | University Student | customize the maximum tasks permitted per day                              | schedule my tasks without burning myself out                                   |
+| v2.0    | University Student | customize the year of my schedule                                          | plan beyond my acedemic years                                                  |
+| v2.0    | University Student | customise the duration to add a certain recurring event                    | adjust it based on the event                                                   |
+| v2.0    | University Student | have reminders for events and deadlines coming soon                        | plan my time accordingly to complete/attend them                               |
+| v2.0    | University Student | have different views of events                                             | so that it is clearer to disntinguish the important ones without having others |
+| v2.0    | University Student | undo my last course action                                                 | reverse accidental changes to the course tracker                               |
```
Comment (alsonsim) [inline docs/DeveloperGuide.md]: The User Stories table contains at least two typos visible in the DG:

    "track the componenets that make up the grades" — should be "components"

    "plan my time accordingly to complete/attend them" — grammatically awkward; should be "plan my time to complete or attend them"

    "so that it is clearer to disntinguish the important ones" — should be "distinguish"
    Should these be corrected? Typos in user stories are minor but reduce the perceived quality of the documentation.

**Review by alsonsim — COMMENTED** [review]
Overall, the DG is well-structured and covers a broad range of features including event management, course tracking, and undo. The UI-to-Data mapping explanation with the ActiveDisplayMap and EventReference tables is a particularly clear and useful design explanation. A few areas for improvement:

    The Instructions for Manual Testing section only covers adding todos — given the breadth of features documented, testing steps for deadlines, events, course tracker, and undo would make this section significantly more useful.

    Several typos appear in the User Stories table ("componenets", "disntinguish") and in the Add Event workflow (nested duplicate method call addRecurringWeeklyEvent(addRecurringWeeklyEvent(...))).

    The Undo feature's scope is limited to course commands only, but no rationale is given for excluding task commands — this could confuse future developers looking to extend the feature.

    The relationship between Storage as a top-level architecture component and as a field inside AppContainer is ambiguous and could benefit from a brief clarification.
    These are minor issues overall — the DG demonstrates good coverage of the system's design and implementation decisions.

File: docs/DeveloperGuide.md [hunk 15]
```diff
@@ -2,37 +2,498 @@
...
 
-## Instructions for manual testing
+### Adding Todos
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+1. Launch the application.
+2. Ensure at least one category exists. If not, create one: `add category School`
+3. Add a todo to a category: `add todo 1 finish tutorial`
+4. Add a todo with priority: `add todo 1 reply email /p 5`
+5. Verify that the todo appears under the School category using: `list category`
```
Comment (Nishchay2576) [inline docs/DeveloperGuide.md]: Great overall work on the DG, all the design considerations look explained and well-reasoned. Could update it a bit to improve consistency.

File: docs/DeveloperGuide.md [hunk 16]
```diff
@@ -2,37 +2,498 @@
...
+2. Input is passed to `CommandParser`
+3. `CommandParser` returns the appropriate `Command` object
+4. `execute(container)` is called
+5. Command updates the model via `AppContainer`
+6. Changes are saved via `Storage`
+7. Output messages are displayed via `UI`
+
+## Implementation
+
+### CategoryList Class Diagram
```
Comment (Nishchay2576) [inline docs/DeveloperGuide.md:135]: I noticed that the level of detail varies across the class diagrams. For instance, the CategoryList diagram successfully omits methods to keep the focus on structural relationships, whereas the Deadline and Storage class diagrams include almost all methods and attributes. Would it be better to standardize the level of abstraction across the guide, perhaps by omitting minor utility methods (like getters/setters and toString()) from all diagrams to consistently highlight only the most important details?

File: docs/DeveloperGuide.md [hunk 17]
```diff
@@ -2,37 +2,498 @@
...
+2. Should be able to hold up to 500 tasks/courses without a noticeable reduction in performance
+3. A user with above average typing speed for regular English text 
+can accomplish most of the tasks faster using commands than using the mouse.
 
 ## Glossary
 
-* *glossary item* - Definition
+* *Mainstream OS* - Windows, Linux, Unix, macOS
+
+## Instructions for Manual Testing
```
Comment (Nishchay2576) [inline docs/DeveloperGuide.md:524]: The "Instructions for Manual Testing" section currently only covers adding a basic Todo. Since the application includes more complex, standout features like the Course Tracker, Undo functionality, and Recurring Events, would it be beneficial to include testing steps for these as well? Adding instructions on how to load sample data and test these specific edge cases would provide better evidence of your ability to document testing procedures.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Nishchay2576) [inline docs/DeveloperGuide.md:41]: In the Architecture section's sequence diagram for the ReorderCommand, specific string inputs and integer arguments (like "reorder todo 1 1 2" and "0, 0, 1") are used. Would replacing these with generic parameter names or variables help the diagram remain accurate and applicable even if the specific command formatting or indexing logic changes in the future?

File: docs/DeveloperGuide.md
Comment (Novgor0d) [inline docs/DeveloperGuide.md:1]: The DG covers the architecture and implementation of UniTasker quite thoroughly with plenty of UML diagrams and explanations given for the features. Overall, it appears polished but could benefit from a few minor tweaks.

File: docs/DeveloperGuide.md (hunk 17 above)
Comment (Novgor0d) [inline docs/DeveloperGuide.md:524]: The Manual Testing section only covers adding `todos`. Should there be test cases for the other major features like `deadlines`, `events`, `recurring events`, and `course commands` which are documented in the DG too?

File: docs/DeveloperGuide.md [hunk 18]
```diff
@@ -2,37 +2,498 @@
...
+| v1.0    | University Student | view all assessments within a course                                       | understand how my course grading is structured                                 |
+| v1.0    | University Student | record my score for an assessment                                          | keep track of my performance in each assessment                                |
+| v2.0    | University Student | delete all marked tasks                                                    | quickly clean up completed work across categories                              |
+| v2.0    | University Student | search for tasks across all categories                                     | quickly find relevant tasks                                                    |
+| v2.0    | University Student | customize the maximum tasks permitted per day                              | schedule my tasks without burning myself out                                   |
+| v2.0    | University Student | customize the year of my schedule                                          | plan beyond my acedemic years                                                  |
```
Comment (Novgor0d) [inline docs/DeveloperGuide.md]: There is a typo in the user stories table "acedemic" should be "academic" (under the v2.0 story about customizing the year).

File: docs/DeveloperGuide.md [hunk 19]
```diff
@@ -2,37 +2,498 @@
...
+
+1. User enters command in terminal
+2. UniTasker reads the input and passes it to CommandParser
+3. CommandParser identifies the command type and creates a corresponding Command object
+4. The Command object calls its execute() method with the AppContainer instance
+5. The Command modifies the appropriate data structures within AppContainer (categories, tasks, calendar)
+6. The Command calls Storage to persist changes to disk
+7. UI components display the result of the operation to the user
+
+**AppContainer component**
```
Comment (Novgor0d) [inline docs/DeveloperGuide.md:53]: The AppContainer section here describes the component and its fields, but there doesn't seem to be a class diagram showing how AppContainer relates to CategoryList, Calendar, Storage, and CourseParser. Given that AppContainer is described as the 'central container' that all commands depend on, wouldn't a class diagram here help a future developer understand these dependencies at a glance? Could a UML diagram be added here?

File: docs/DeveloperGuide.md [hunk 20]
```diff
@@ -2,37 +2,498 @@
...
+This design improves modularity and keeps command logic simple, while allowing `CategoryList` to coordinate operations affecting all three task lists.
+
+### Deadline Class Diagram
+
+The figure below illustrates the relationship between Deadline class and the following classes: Task, Timed, Calendar, DateUtils, DeadlineList, TaskList. 
+
+![Deadline Class Diagram](pictures/deadlineClassDiagram.png)
+
+The `Deadline` class consists of the following members:
+- `logger` - private logger instance used for fin-grained diagnostic logging on object creation
```
Comment (Novgor0d) [inline docs/DeveloperGuide.md:206]: There appears to be a typo here, "fin-grained diagnostic logging" should this be "fine-grained"?

File: docs/DeveloperGuide.md [hunk 21]
```diff
@@ -2,37 +2,498 @@
...
 
-## Design & implementation
+## Design
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+This section describes the design and implementation of the key components of UniTasker. UML diagrams are provided for highlighted classes and interaction flows
 
+### Architecture
+
+![MainArchitecture](pictures/MainArchitecture.png)
```
Comment (Novgor0d) [inline docs/DeveloperGuide.md:19]: In the MainArchitecture diagram perhaps there could be a label explaining what the arrow directions represent for better clarity? (e.g., does an arrow mean "depends on", "calls", "contains", etc.?)
image

File: docs/DeveloperGuide.md
Comment (nicoleroot) [inline docs/DeveloperGuide.md:1]: Generally well-structured with components clearly divided across different packages. Most features have sequence diagrams paired with explanations, and the inheritance and composition relationships are shown. But I feel like manual testing appendix can be more detailed for an actual user to test.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (nicoleroot) [reply docs/DeveloperGuide.md:41]: Agree

File: docs/DeveloperGuide.md [hunk 22]
```diff
@@ -2,37 +2,498 @@
...
+1. User enters command in terminal
+2. UniTasker reads the input and passes it to CommandParser
+3. CommandParser identifies the command type and creates a corresponding Command object
+4. The Command object calls its execute() method with the AppContainer instance
+5. The Command modifies the appropriate data structures within AppContainer (categories, tasks, calendar)
+6. The Command calls Storage to persist changes to disk
+7. UI components display the result of the operation to the user
+
+**AppContainer component**
+
```
Comment (nicoleroot) [inline docs/DeveloperGuide.md:54]: Will it be better if class diagram for AppContainer is provided?

File: docs/DeveloperGuide.md (hunk 15 above)
Comment (nicoleroot) [reply docs/DeveloperGuide.md]: Agree
